src/main/scala/is/hail/utils/IntervalTree.scala

```diff
@@ -163,10 +176,13 @@ case class IntervalTreeNode[U](i: Interval,
   right: Option[IntervalTreeNode[U]],
   minimum: Any, maximum: Any, value: U) extends Traversable[(Interval, U)] {
 
+  override val size: Int =
+    left.map(_.size).getOrElse(0) + right.map(_.size).getOrElse(0) + 1
+
   def contains(pord: ExtendedOrdering, position: Any): Boolean = {
-    pord.gteq(position, minimum) && pord.lteq(position, maximum) &&
+    pord.lteq(minimum, position) && pord.gteq(maximum, position) &&
```

Comment:
Were there bugs here?  This change looks like a no-op.  This code is slightly complicated so that makes me a little nervous.

Reply:
it is now---it wasn't before.

This change means that you can consistently compare things with select orderings, so the "interval" RV and the "position" RV could have different types (as long as the ordering knew how to interpret the fields) when I was doing region value stuff.

Reply:
Ah, I see.  Let's undo them then.

Yeah, having comparisons take different types is dicey business.  Honestly, I don't like that about our UnsafeOrdering.  One has to be quite careful.